storage/innobase/btr/btr0cur.cc

```diff
@@ -3308,16 +3308,15 @@ btr_cur_ins_lock_and_undo(
 			GAP-locking we mark that this transaction
 			is using unique key scan here. */
 			if ((type & (DICT_CLUSTERED | DICT_UNIQUE)) == DICT_UNIQUE
-			    && trx->is_wsrep()
-			    && wsrep_thd_is_BF(trx->mysql_thd, false)) {
-				trx->wsrep_UK_scan= true;
+			    && trx->is_wsrep_BF()) {
+				trx->wsrep |= trx_t::TRX_WSREP_UK_SCAN;
```

Comment:
Instead of modifying `trx->wsrep` directly, should we have `trx->begin_UK_scan()` and `trx->end_UK_scan()` for bit operations to improve readability and reduce chance of error?

Reply:
Sure.